### PR TITLE
Use FileScoped namespaces and refactor tests.

### PR DESCRIPTION
**Pull request by jaliyaudagedara**
Use File Scoped namespaces
Proper use of `var`
Refactor Tests

### CODE REVIEW

File: samples/complexapp/complexapp/Program.cs [hunk 1]
```diff
@@ -1,19 +1,17 @@
-﻿using System;
-using libbar;
+﻿using libbar;
 using libfoo;
 using static System.Console;
 
-namespace complexapp
+namespace complexapp;
```
Comment (mthalman) [inline samples/complexapp/complexapp/Program.cs:5]: Even better than using file-scoped namespace would be to use top-level statements for Program.cs.

File: samples/complexapp/tests/StringUtilsTests.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+namespace tests;
+
+public class StringUtilsTests
+{
+    [Fact]
+    public void ReverseString_WhenInputIsNull_ThrowsCorrectException()
+    {
+        string inputString = null;
+        Assert.Throws<ArgumentNullException>(() => StringUtils.ReverseString(inputString));
+    }
```
Comment (mthalman) [inline samples/complexapp/tests/StringUtilsTests.cs:14]: The addition of this extra test is causing a test failure in the expected number of tests. These values need to be updated:

https://github.com/dotnet/dotnet-docker/blob/609446a3686940137628bbb67404f427294813ae/tests/Microsoft.DotNet.Docker.Tests/SampleImageTests.cs#L129-L130

File: samples/complexapp/complexapp/Program.cs (hunk 1 above)
Comment (jaliyaudagedara) [reply samples/complexapp/complexapp/Program.cs]: Of course, updated

File: samples/complexapp/tests/StringUtilsTests.cs (hunk 2 above)
Comment (jaliyaudagedara) [reply samples/complexapp/tests/StringUtilsTests.cs:14]: Updated

**Review by mthalman — APPROVED** [review]

**Pull request merged**